packages/cloud-cognitive/src/components/APIKeyModal/APIKeyModal.js

```diff
@@ -236,174 +282,83 @@ export let APIKeyModal = forwardRef(
         </ModalBody>
         <ModalFooter className={`${blockClass}__footer`}>
           <Button type="button" kind="secondary" onClick={onBackHandler}>
-            {getSecondaryButtonText()}
+            {secondaryButtonText}
           </Button>
           <Button
-            {...(apiKeyLoaded ? { renderIcon: Copy16 } : {})}
+            {...(apiKeyLoaded ? copyButtonProps : {})}
             type="submit"
             kind="primary"
             onClick={submitHandler}
-            disabled={isPrimaryButtonDisabled()}>
-            {getPrimaryButtonText()}
+            disabled={isPrimaryButtonDisabled}>
+            {primaryButtonText}
           </Button>
         </ModalFooter>
       </ComposedModal>
     );
   }
 );
 
+const customStepsRequiredProps = isRequiredIf(
```

Comment:
Using PropTypes.string.isRequired.if directly in propTypes will add an * to the field in storybook documentation. Also worth adding a comment/note to the comment in propTypes e.g. NOTE: This prop is only required if customSteps are supplied.